### PR TITLE
Adding Maven publish WF 

### PR DESCRIPTION
**Pull request by khushbr**
**Is your feature request related to a problem? Please provide an existing Issue # , or describe.**
Adding Maven publish WF, following https://opensearch.org/blog/opensearch-plugin-zips-now-in-maven-repo/  and https://github.com/opensearch-project/opensearch-build/blob/main/ONBOARDING.md

```
% ./gradlew publishToMavenLocal
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
...
BUILD SUCCESSFUL in 1s
3 actionable tasks: 2 executed, 1 up-to-date

% ls /Users/khushbr/.m2/repository/org/opensearch/performance-analyzer-commons/1.0-SNAPSHOT
maven-metadata-local.xml			performance-analyzer-commons-1.0-SNAPSHOT.jar	performance-analyzer-commons-1.0-SNAPSHOT.pom
```

Following https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing, set url to /Users/khushbr/test/
```
% ls /Users/khushbr/test/org/opensearch/performance-analyzer-commons/1.0.0.0-SNAPSHOT
maven-metadata.xml							performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar.sha256
maven-metadata.xml.md5							performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar.sha512
maven-metadata.xml.sha1							performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom
maven-metadata.xml.sha256						performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom.md5
maven-metadata.xml.sha512						performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom.sha1
performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar		performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom.sha256
performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar.md5		performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom.sha512
performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar.sha1
```

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/performance-analyzer-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaiksaya** [comment]
Can you testing the changes? See https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing for the steps

File: .github/workflows/maven-publish.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        with:
+          role-to-assume: ${{ secrets.PUBLISH_SNAPSHOTS_ROLE }}
+          aws-region: us-east-1
+      - name: publish snapshots to maven
+        run: |
+          export SONATYPE_USERNAME=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-username --query SecretString --output text)
+          export SONATYPE_PASSWORD=$(aws secretsmanager get-secret-value --secret-id maven-snapshots-password --query SecretString --output text)
+          echo "::add-mask::$SONATYPE_USERNAME"
+          echo "::add-mask::$SONATYPE_PASSWORD"
+          ./gradlew publish
```
Comment (gaiksaya) [inline .github/workflows/maven-publish.yml:33]: Should only publish to snapshots repository

Comment (khushbr) [reply .github/workflows/maven-publish.yml:33]: @gaiksaya Wondering how is this different from `publishToMavenLocal` ? 

Following https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing, set url to 
`/Users/khushbr/test/`
```

% ls /Users/khushbr/test/org/opensearch/performance-analyzer-commons/1.0-SNAPSHOT
maven-metadata.xml						performance-analyzer-commons-1.0-20230523.200158-1.jar		performance-analyzer-commons-1.0-20230523.200158-1.pom
maven-metadata.xml.md5						performance-analyzer-commons-1.0-20230523.200158-1.jar.md5	performance-analyzer-commons-1.0-20230523.200158-1.pom.md5
maven-metadata.xml.sha1						performance-analyzer-commons-1.0-20230523.200158-1.jar.sha1	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha1
maven-metadata.xml.sha256					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha256	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha256
maven-metadata.xml.sha512					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha512	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha512
```

**Comment by khushbr** [comment]
> Can you testing the changes? See https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing for the steps

@gaiksaya Wondering how is this different from `publishToMavenLocal` ? 

Following https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing, set url to 
`/Users/khushbr/test/`
```

% ls /Users/khushbr/test/org/opensearch/performance-analyzer-commons/1.0-SNAPSHOT
maven-metadata.xml						performance-analyzer-commons-1.0-20230523.200158-1.jar		performance-analyzer-commons-1.0-20230523.200158-1.pom
maven-metadata.xml.md5						performance-analyzer-commons-1.0-20230523.200158-1.jar.md5	performance-analyzer-commons-1.0-20230523.200158-1.pom.md5
maven-metadata.xml.sha1						performance-analyzer-commons-1.0-20230523.200158-1.jar.sha1	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha1
maven-metadata.xml.sha256					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha256	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha256
maven-metadata.xml.sha512					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha512	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha512
```

**Comment by gaiksaya** [comment]
> > Can you testing the changes? See https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing for the steps
> 
> @gaiksaya Wondering how is this different from `publishToMavenLocal` ?
> 
> Following https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing, set url to `/Users/khushbr/test/`
> 
> ```
> 
> % ls /Users/khushbr/test/org/opensearch/performance-analyzer-commons/1.0-SNAPSHOT
> maven-metadata.xml						performance-analyzer-commons-1.0-20230523.200158-1.jar		performance-analyzer-commons-1.0-20230523.200158-1.pom
> maven-metadata.xml.md5						performance-analyzer-commons-1.0-20230523.200158-1.jar.md5	performance-analyzer-commons-1.0-20230523.200158-1.pom.md5
> maven-metadata.xml.sha1						performance-analyzer-commons-1.0-20230523.200158-1.jar.sha1	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha1
> maven-metadata.xml.sha256					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha256	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha256
> maven-metadata.xml.sha512					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha512	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha512
> ```

Maven local also has generates different kind of metadata.xml. Also above artifacts are missing "SNAPSHOTS" keyword which might lead to failures during upload

**Comment by khushbr** [comment]
> > > Can you testing the changes? See https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing for the steps
> > 
> > 
> > @gaiksaya Wondering how is this different from `publishToMavenLocal` ?
> > Following https://github.com/opensearch-project/opensearch-plugins/blob/main/WORKFLOWS.md#testing, set url to `/Users/khushbr/test/`
> > ```
> > 
> > % ls /Users/khushbr/test/org/opensearch/performance-analyzer-commons/1.0-SNAPSHOT
> > maven-metadata.xml						performance-analyzer-commons-1.0-20230523.200158-1.jar		performance-analyzer-commons-1.0-20230523.200158-1.pom
> > maven-metadata.xml.md5						performance-analyzer-commons-1.0-20230523.200158-1.jar.md5	performance-analyzer-commons-1.0-20230523.200158-1.pom.md5
> > maven-metadata.xml.sha1						performance-analyzer-commons-1.0-20230523.200158-1.jar.sha1	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha1
> > maven-metadata.xml.sha256					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha256	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha256
> > maven-metadata.xml.sha512					performance-analyzer-commons-1.0-20230523.200158-1.jar.sha512	performance-analyzer-commons-1.0-20230523.200158-1.pom.sha512
> > ```
> 
> Maven local also has generates different kind of metadata.xml. Also above artifacts are missing "SNAPSHOTS" keyword which might lead to failures during upload

Thank you clarifying, @gaiksaya . Fixed the issue.

**Comment by gaiksaya** [comment]
Thanks! Can you also update the comment with new snaphots that are being published? Also, just confirming `org.opensearch.performance-analyzer-commons` is the right maven coordinates right?
Adding @prudhvigodithi for one more pair of eyes

**Comment by prudhvigodithi** [comment]
Does this plugin also generate a Zip file to install apart for the jar?
@khushbr

**Comment by khushbr** [comment]
> Thanks! Can you also update the comment with new snaphots that are being published? Also, just confirming `org.opensearch.performance-analyzer-commons` is the right maven coordinates right? Adding @prudhvigodithi for one more pair of eyes

@gaiksaya The [PR description](https://github.com/opensearch-project/performance-analyzer-commons/pull/11#issue-1722672036) has been updated with the latest snpashots getting published. Adding here as well.

And yes, `org.opensearch.performance-analyzer-commons` is the right maven location. 

```
% ls /Users/khushbr/test/org/opensearch/performance-analyzer-commons/1.0.0.0-SNAPSHOT
maven-metadata.xml							performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar.sha256
maven-metadata.xml.md5							performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar.sha512
maven-metadata.xml.sha1							performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom
maven-metadata.xml.sha256						performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom.md5
maven-metadata.xml.sha512						performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom.sha1
performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar		performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom.sha256
performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar.md5		performance-analyzer-commons-1.0.0.0-20230523.202721-1.pom.sha512
performance-analyzer-commons-1.0.0.0-20230523.202721-1.jar.sha1

```

**Comment by khushbr** [comment]
> Does this plugin also generate a Zip file to install apart from the jar? As did notice the following is commented out
> 
> ```
> distZip {
>     archiveName "performance-analyzer-commons-${version}.zip"
> }
> ```
> 
> @khushbr

@prudhvigodithi Only the source JAR is required from commons. Curious, any reason to publish a zip here that you are aware?

**Comment by prudhvigodithi** [comment]
Got it, plugins can be installed with `zip` file, so we are publishing the plugin zips also to maven along with jars, please check https://opensearch.org/blog/opensearch-plugin-zips-now-in-maven-repo/.

**Pull request merged**